main.go

```diff
@@ -31,6 +31,398 @@ const bwSuccessPrefix = "✓"
 const bwFailPrefix = "𐄂"
 const bwErrorPrefix = "!"
 
+/**
+ * Call function and, when it is possible display spinner with some progress message
+ */
+func showProgress(progressMessage string, isColorful bool, function func() error) error {
+	var s *spinner.Spinner
+	if isColorful {
+		s = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		s.Suffix = progressMessage
+		s.Start()
+	}
+
+	err := function()
+
+	if isColorful {
+		s.Stop()
+	}
+
+	return err
+}
+
+/**
+ * Try to get color preferences form context
+ */
+func getColorPreferences(ctx *cli.Context) (string, string, string, bool) {
```

Comment:
This could be a good opportunity to use named return values.

```suggestion
func getColorPreferences(ctx *cli.Context) (successPrefix string, failPrefix string, errorPrefix string, isColorful bool) {
```

This way, you can remove the variable declarations in lines 60 through 63. After setting the values, line 75 can call an empty `return`. This removes ambiguity around the order of return values.

Reply:
Yeah, I know that Go has this functionality, but old C coder tents to use old-fashion style.

Reply:
Yeah, I agree. It is already implemented in the way that you suggested. I only tried to explain, why I coded it originally in that way.